reference/usage.v1.yaml

```diff
@@ -0,0 +1,142 @@
+openapi: 3.0.0
+info:
+  title: Usage API
+  version: '1.0'
+  description: |
+    This is an API provided by the Learning Application in the SEM ecosystem, and is designed to enable a learning application to make information about usage of a product by a teacher or student to be confirmed to the Marketplace
+
+    # Events
+
+    It is envisaged that the service implementing this API, also implements the generic [Events](https://stichtingsem.stoplight.io/docs/sem-technology-prototype/reference/events.v1.yaml) and [Webhooks](https://stichtingsem.stoplight.io/docs/sem-technology-prototype/reference/webhooks.v1.yaml) API to enable subscription to changes vs polling.
+
+
+    Events generated by the Usage API are connected to (first) use of the product.
+     - `la.usage.first` - First time
+     - `la.usage.ongoing` - Regular usage
+  contact:
+    name: SEM
+    url: 'https://stichtingsem.org/'
+servers:
+  - url: 'https://example.stichtingsem.org/api'
+    description: usage
+paths:
+  '/usage/{entitlementid}':
+    parameters:
+      - schema:
+          type: string
+        name: entitlementid
+        in: path
+        required: true
+    get:
+      summary: Get usage for a specific entitlement
+      tags:
+        - usage
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/usage'
+            application/xml:
+              schema:
+                type: object
+                properties: {}
+      operationId: get-usage-by-entitlement-id
+      description: |-
+        Retrieve the full usage based on a entitlement-id, needs to confirm that the caller is able to access either based on their provider information (they can only see their own usage) in the case of the provider scope.
+        This API has pagination, using `start` and `limit`, with an optional `status` filter.
+      security:
+        - APIKey:
+            - la.usage.first
+            - la.usage.ongoing
+      parameters:
+        - in: query
+          name: status
+          description: |
+            Filter by status
+          schema:
+            type: string
+            enum:
+              - uniqueActivation
+              - weeklyActivation
```

Comment:
These I struggle with - I had only envisaged usage at the `unique` level, not with varying aggregations.  What would the usageDate be for 'weekly' or 'monthly' values?  I think these aggregates should be different endpoints with a different shape.

I also think usage should be reflective of the entitlement - if an entitlement is at School level, we reflect usage at that level.  Otherwise we are in a situation where we are sending more information about commercial usage than was shared with us (e.g. 100 for School X, and instead of receiving 98 for School X, you receive Student 1 ... 98).

Is there a support use case here where you really need to see (as MP) that student X specifically used the product? (as you can also see this in the LA itself).

Reply:
@cliftonc : From my perspective weekly or monthly is just an aggregation of all unique activations

Reply:
Understood - but I don't know why this API provides it in this way, it triggers more questions.  

If we think it is necessary for the LA to aggregate it for the MP (it can aggregate it itself if we send it the underlying data), then we should put it in a separate endpoint to then be able to clarify the exact range over which the aggregation applies (start and end dates).

I personally don't see the need, and would prefer to just send usage info in a 'unique' form related to the original entitlement and figure out how to make that efficient - that is simplest if we log every access, and send them as they happen, but this may result in a large number of events (albeit small ones).

Reply:
I don't think we should report on school level. Reporting on individual level end all discussion on how many students have used the product in the entitlement. Both MP and LA have the same view on which students are active. so for transparency reasons.

I think it is up to the LA to decide how to report. First time usage seems obvious, but if a entitlement is by mistake activated you can see that the student only used the product once or twice.
Sending all uniqueActivations seems like a lot, (and needless) so I added the option to report every week or month that a product is used. That gives a lot of information to see if the product is active.

The API supports it, I think the initialActivation is mandatory, the other optional?